aea/mail/base.py

```diff
@@ -472,6 +474,11 @@ def _stop(self):
             logger.debug("Using default connection: {}".format(self.default_connection))
             connection = self.default_connection
 
+        if envelope.protocol_id not in connection.supported_protocols:
+            logger.warning("Connection {} does not support protocol {}. Cannot send the message."
+                           .format(connection.connection_id, envelope.protocol_id))
+            return
```

Comment:
Observe that, in case the protocol id of the outgoing envelope is not supported  by the chosen connection, we just log a warning message. That might not be always ideal.

Reply:
Ok, for now that is fine.

Reply:
Alternative is to raise an error?

Reply:
An alternative is to let the user specifying a callback to handle this case.